internal/controller/promotions/promotions.go

```diff
@@ -458,6 +457,7 @@ func (r *reconciler) promote(
 	if err != nil {
 		return nil, err
 	}
+	newStatus = newStatus.WithFreight(&nextFreight)
```

Comment:
Nit: Personally, I'm not sure I would delegate such a trivial assignment as `newStatus.Freight = nextFreight` to this new `WithFreight()` function.

Reply:
I thought the same but figured we could sort it with the other suggestions we would tackle ourselves.